operator/v1/types_csi_cluster_driver.go

```diff
@@ -118,25 +121,136 @@ const (
 type CSIDriverConfigSpec struct {
 	// driverType indicates type of CSI driver for which the
 	// driverConfig is being applied to.
-	//
-	// Valid values are:
-	//
-	// * vSphere
-	//
-	// Allows configuration of vsphere CSI driver topology.
-	//
-	// ---
+	// Valid values are: AWS, Azure, GCP, vSphere
 	// Consumers should treat unknown values as a NO-OP.
-	//
 	// +kubebuilder:validation:Required
 	// +unionDiscriminator
 	DriverType CSIDriverType `json:"driverType"`
 
+	// AWS is used to configure the AWS CSI driver.
+	// +optional
+	AWS *AWSCSIDriverConfigSpec `json:"aws,omitempty"`
+
+	// Azure is used to configure the Azure CSI driver.
+	// +optional
+	Azure *AzureCSIDriverConfigSpec `json:"azure,omitempty"`
+
+	// GCP is used to configure the GCP CSI driver.
+	// +optional
+	GCP *GCPCSIDriverConfigSpec `json:"gcp,omitempty"`
+
 	// vsphere is used to configure the vsphere CSI driver.
 	// +optional
 	VSphere *VSphereCSIDriverConfigSpec `json:"vSphere,omitempty"`
 }
 
+// AWSCSIDriverConfigSpec defines properties that can be configured for the AWS CSI driver.
+type AWSCSIDriverConfigSpec struct {
+	// kmsKey sets the cluster default storage class to encrypt volumes with a user-defined KMS key,
+	// rather than the default KMS key used by AWS.
+	// +optional
+	KMSKey AWSKMSKey `json:"kmsKey,omitempty"`
+}
+
+// AWSKMSKey contains the component parts that comprise an ARN for a KMS key.
+// According to https://docs.aws.amazon.com/service-authorization/latest/reference/list_awskeymanagementservice.html
+// the format for a KMS key ARN is:
+// arn:${Partition}:kms:${Region}:${Account}:key/${KeyId}
+// When constructing the ARN, the partition can be determined from the region.
+type AWSKMSKey struct {
```

Comment:
There was [some discussion](https://github.com/openshift/enhancements/pull/1163#discussion_r1087055672) on splitting up the ARN string in the enhancement, but it raised some problems so we agreed on [using the ARN string](https://github.com/openshift/enhancements/blob/ac2729f61cb593a82cfe8a7bb771e7c1d5043fcb/enhancements/installer/storage-class-encrypted.md#aws) like we use for [root volumes](https://access.redhat.com/solutions/4827341) today.

Reply:
Fixed.